### PR TITLE
Add a VS solution/project for winuser

### PR DESCRIPTION
**Pull request by csujedihy**
Add a VS solution/project for winuser so that consumers do not have to create redundant project files to build cxplat.

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -417,7 +417,17 @@ if ($Platform -eq "winkernel") {
     if (!$ConfigureOnly) {
         # Build the code.
         Log "Building..."
-        msbuild cxplat.kernel.sln /m /p:Configuration=$Config /p:Platform=$Arch
+        msbuild cxplat.kerenl.sln /m /p:Configuration=$Config /p:Platform=$Arch
```
Comment (nigriMSFT) [inline scripts/build.ps1]: Unintentional typo "cxplat.kerenl.sln"?

Comment (csujedihy) [reply scripts/build.ps1]: oops

File: scripts/build.ps1 [hunk 2]
```diff
@@ -69,7 +69,7 @@ param (
     [string]$Arch = "",
 
     [Parameter(Mandatory = $false)]
-    [ValidateSet("gamecore_console", "uwp", "windows", "linux", "macos", "android", "ios", "winkernel")] # For future expansion
+    [ValidateSet("gamecore_console", "uwp", "windows", "linux", "macos", "android", "ios", "winkernel", "windows-vs")] # For future expansion
```
Comment (nigriMSFT) [inline scripts/build.ps1]: Exposing this new build option as a new "Platform" might be confusing. windows user mode built with "windows" platform will be equivalent to windows user mode built with "windows-vs" platform. They will both be usable for win32 apps.

I'd suggest introducing a "BuildToolSet" or similar that allows you to specify a specific build tool set within a platform.

.\build.ps1 -Platform windows -BuildToolSet vs
or 
.\build.ps1 -Platform windows -BuildToolSet cmake

Comment (csujedihy) [reply scripts/build.ps1]: good idea

Comment (nigriMSFT) [reply scripts/build.ps1]: There is already a Clang switch that is related to this. Hmm, I'm not exactly sure what the best way is to expose all of these related options. I think we can do better than extending "Platform" though

Comment (csujedihy) [reply scripts/build.ps1]: Added BuildToolSet

File: scripts/build.ps1 [hunk 3]
```diff
@@ -409,15 +427,29 @@ function CMake-Build {
...
-        # Build the code.
-        Log "Building..."
-        msbuild cxplat.kernel.sln /m /p:Configuration=$Config /p:Platform=$Arch
+        if (!$ConfigureOnly) {
+            # Build the code.
+            Log "Building..."
+            msbuild cxplat.kernel.sln /m /p:Configuration=$Config /p:Platform=$Arch
+        }
+    } elseif ($Platform -eq "windows") {
+        # Restore Nuget packages.
```
Comment (nigriMSFT) [inline scripts/build.ps1:442]: Just set a $solutionFile variable based on $Platform value and we can re-use the same code block for both winkernel and windows vs build?

**Review by nigriMSFT — APPROVED** [review]

**Pull request merged**